src/3d/qgswindow3dengine.h

```diff
@@ -78,6 +78,8 @@ class _3D_EXPORT QgsWindow3DEngine : public QgsAbstract3DEngine
     QSize size() const override;
     QSurface *surface() const override;
 
+    //! Sets the size of the window (useful to fix frame graph parameters)
+    void setSize( int width, int height );
```

Comment:
We already have setSize() in QgsOffscreen3DEngine - so I would suggest to declare `setSize(QSize)` as virtual.

Also please have a look at the setSize() implementation in the offscreen engine... how about setting camera aspect ratio, don't we need to set it for window 3d engine as well?